src/main/java/org/apache/sysml/runtime/codegen/LibSpoofPrimitives.java

```diff
@@ -1914,7 +1915,40 @@ public static void vectGreaterequalAdd(double bval, double[] a, double[] c, int[
 		//invariant to the ordering of inputs
 		return vectLessWrite(b, a, bix, bi, ai, blen, len);
 	}
-	
+
+	//********************* Logical Bitwise Operators ********************//
+
+	/**
+	 * BitwiseAnd Operator
+	 *
+	 * @return
+	 */
+
+	//1. dense vector vs. scalar
+	public static double[] vectBitwandWrite(double[] a, double bval, int ai, int len) {
+		double[] c = allocVector(len, false);
+		for( int j = 0; j < len; j++ )
+			c[j] = (double)( UtilFunctions.toInt(a[ai+j]) & UtilFunctions.toInt(bval) );
+		return c;
+	}
+
+	//2. scalar vs. dense vector
+	public static double[] vectBitwandWrite(double bval, double[] a, int ai, int len) {
+		double[] c = allocVector(len, false);
+		for( int j = 0; j < len; j++ )
+			c[j] = ( UtilFunctions.toInt(bval) & UtilFunctions.toInt(a[ai+j]) );
+		return c;
+	}
+
+	//3. dense vector vs. dense vector
+	public static double[] vectBitwandWrite(double[] a, double[] b, int ai, int bi, int len) {
+		double[] c= allocVector(len, false);
+		for( int j = 0; j < len; j++ )
+			c[j] = (double)( UtilFunctions.toInt(a[ai+j]) & UtilFunctions.toInt(b[bi+j]));
```

Comment:
Same casting as described above.

Reply:
Hi @mboehm7 ,
tried to cast for int as `(int)a[ai+j]`, without success. 
1. `Math.floor(a[ai+j])` is giving correct result.
2. It can be that, I am doing some mistake else where.